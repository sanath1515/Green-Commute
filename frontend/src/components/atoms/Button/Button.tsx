import React from "react";
import { Button as MuiButton } from "@material-ui/core";

export type ButtonProps = {
  name: string;
  variant?: "contained" | "text" | "outlined";
  color?: "primary" | "secondary" | "inherit";
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  className?: string;
  startIcon?: JSX.Element;
};

export const Button = (props: ButtonProps) => {
  return (
    <>
      <MuiButton
        variant={props.variant}
        onClick={props.onClick}
        color={props.color}
        className={props.className}
        startIcon={props.startIcon}
        {...props}
      >
        {props.name}
      </MuiButton>
    </>
  );
};
