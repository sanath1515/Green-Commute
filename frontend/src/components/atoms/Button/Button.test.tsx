/**
 * @jest-environment jsdom
 */
import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./Button";

describe("Button Component", () => {
  const onClick = jest.fn();

  test("Matches Snapshot", () => {
    const { asFragment } = render(<Button name="hello" onClick={onClick} />);
    expect(asFragment.name).toMatchSnapshot();
  });

  it("should get button text", () => {
    render(<Button name="hello" onClick={onClick} />);
    expect(screen.getByText("hello")).toBeInTheDocument();
  });

  it("should call onclick", () => {
    const { getByText } = render(<Button name="hello" onClick={onClick} />);
    fireEvent.click(getByText("hello"));
    expect(onClick).toHaveBeenCalled();
  });
});
