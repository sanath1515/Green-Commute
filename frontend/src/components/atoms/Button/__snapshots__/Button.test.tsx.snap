// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Matches Snapshot 1`] = `"asFragment"`;
